app/controllers/candidate_interface/sign_up_controller.rb

```diff
@@ -5,13 +5,20 @@ def new
     end
 
     def create
-      @candidate = Candidate.new(candidate_params)
+      @candidate = Candidate.find_by(candidate_params)
 
-      if @candidate.save
-        MagicLinkSignUp.call(candidate: @candidate)
-        render :show
+      if @candidate
+        MagicLinkSignIn.call(candidate: @candidate)
+        render 'candidate_interface/sign_in/show'
       else
-        render :new
+        @candidate = Candidate.new(candidate_params)
+
+        if @candidate.save
+          MagicLinkSignUp.call(candidate: @candidate)
+          render :show
+        else
+          render :new
+        end
```

Comment:
I can think of an alternative implementation with [find_or_initialize_by](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_initialize_by):

```ruby
@candidate = Candidate.find_or_initialize_by(candidate_params)

if @candidate.persisted?
  MagicLinkSignIn.call(candidate: @candidate)
  render 'candidate_interface/sign_in/show'
elsif @candidate.save
  MagicLinkSignUp.call(candidate: @candidate)
  render :show
else
  render :new
end
```

What do you think? I'm happy with either one!

Reply:
what is the difference between 
`render 'candidate_interface/sign_in/show'`
and
`render :show`
?

Reply:
> I can think of an alternative implementation with [find_or_initialize_by](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_initialize_by):
> 
> ```ruby
> @candidate = Candidate.find_or_initialize_by(candidate_params)
> 
> if @candidate.persisted?
>   MagicLinkSignIn.call(candidate: @candidate)
>   render 'candidate_interface/sign_in/show'
> elsif @candidate.save
>   MagicLinkSignUp.call(candidate: @candidate)
>   render :show
> else
>   render :new
> end
> ```
> 
> What do you think? I'm happy with either one!

Happy to go with this, it's a single if statement rather than my nested and doesn't require to assign `@candidate` if it's found. 👍

I'll update!

Reply:
I think this is because this code is within the SignUpController. 
So `render :show` would render the SignUp show rather than the SignIn show. (Please correct me if I'm wrong 😅)

Seeing as there is sign up and sign in logic here should we rename the controller? 🤔

Reply:
> what is the difference between
> `render 'candidate_interface/sign_in/show'`
> and
> `render :show`
> ?

Currently there is no difference but I imagined after we update the content for all the sign up/in pages and emails we will refactor and only have a single "Check your email" page.

I used the `'candidate_interface/sign_in/show'` to emphasise that from then on it's just the sign in route and a thing to refactor when we do the "improving the content" card. Might be a bit of a silly reason when I think about it now 😅.

Reply:
> I think this is because this code is within the SignUpController.
> So `render :show` would render the SignUp show rather than the SignIn show. (Please correct me if I'm wrong 😅)
> 
> Seeing as there is sign up and sign in logic here should we rename the controller? 🤔

I think you're right, some refactoring needs to be done. I'm thinking we get this through first and then we can do it in when we do the https://trello.com/c/xBMMWn5J/73-improve-content-in-sign-in-up-pages-and-emails card.

Reply:
Yup, we can always change our minds and merge controllers if they are coupled anyway and deal with the same concern. Personally I think it's okay for now :)